### PR TITLE
viper website docs

### PR DESCRIPTION
**Pull request by ajm188**
A large portion of this was extracted from parts of the doc written in https://github.com/vitessio/vitess/pull/11456

~~I still need to update the markdown links to make them work here, but the text stands on its own~~

### CODE REVIEW

File: content/en/docs/17.0/reference/viper/basic.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## `Get` funcs
+
+In most cases, module authors will not need to specify a `GetFunc` option, since, if not provided, `viperutil` will do its best to provide a sensible default for the given type `T`.
+
+This requires a fair amount of `reflect`ion code, which we won't go into here, and unfortunately cannot support even all primitive types (notably, array (not slice!!) types).
+In these cases, the `GetFuncForType` will panic, allowing the module author to catch this during testing of their package.
+They may then provide their own `GetFunc`.
+
+The full suite of types, both supported and panic-inducing, are documented by way of unit tests in [`go/viperutil/get_func_test.go`](../../go/viperutil/get_func_test.go).
```
Comment (mattlord) [inline content/en/docs/17.0/reference/viper/basic.md]: This link will not work: https://deploy-preview-1482--vitess.netlify.app/docs/17.0/reference/viper/basic/#get-funcs

File: content/en/docs/17.0/reference/viper/overview.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+title: Viper Overview
+weight: 1
+---
+
+Vitess v17 introduced [`viper`][viper], a library to provide unified configuration management, to the project.
```
Comment (mattlord) [inline content/en/docs/17.0/reference/viper/overview.md:6]: What is this supposed to be? It's not a link: https://deploy-preview-1482--vitess.netlify.app/docs/17.0/reference/viper/overview/

File: content/en/docs/17.0/reference/viper/overview.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+title: Viper Overview
```
Comment (mattlord) [inline content/en/docs/17.0/reference/viper/overview.md]: None of the other page titles have `Viper` in it. So I would suggest using `Overview` instead.

File: content/en/docs/17.0/reference/viper/dynamic_values.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Values can be configured to be either static or dynamic.
+Static values are loaded once at startup (more precisely, when `viperutil.LoadConfig` is called), and whatever value is loaded at the point will be the result of calling `Get` on that value for the remainder of the process's lifetime.
+Dynamic values, conversely, may respond to config changes.
+
+In order for dynamic configs to be truly dynamic, `LoadConfig` must have found a config file (as opposed to pulling values entirely from defaults, flags, and environment variables).
+If this is the case, a second viper shim, which backs the dynamic registry, will start a watch on that file, and any changes to that file will be reflected in the `Get` methods of any values configured with `Dynamic: true`.
+
+**An important caveat** is that viper on its own is not threadsafe, meaning that if a config reload is being processed while a value is being accessed, a race condition can occur.
+To protect against this, the dynamic registry uses a second shim, [`sync.Viper`](../../go/viperutil/internal/sync/sync.go).
```
Comment (mattlord) [inline content/en/docs/17.0/reference/viper/dynamic_values.md]: This link does not work: https://deploy-preview-1482--vitess.netlify.app/docs/17.0/reference/viper/dynamic-values/

**Review by mattlord — COMMENTED** [review]
A few minor suggestions and requests. Beyond that, we need to rebase this on the `prod` branch and copy the new files to the 18.0 docs as well.

File: content/en/docs/17.0/reference/viper/overview.md (hunk 2 above)
Comment (ajm188) [reply content/en/docs/17.0/reference/viper/overview.md:6]: Turns out .......

```
$ git switch andrew/viper 
Switched to branch 'andrew/viper'
Your branch is ahead of 'origin/andrew/viper' by 1 commit.
  (use "git push" to publish your local commits)
$ git show
commit 9bb689c6bd824213c111ba888b381972e1cd4e78 (HEAD -> andrew/viper)
Author: Andrew Mason <andrew@planetscale.com>
Date:   Wed May 24 09:10:25 2023 -0400

    fix links
    
    Signed-off-by: Andrew Mason <andrew@planetscale.com>
```


🤦 sorry about that!

File: content/en/docs/17.0/reference/viper/overview.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+- Default values.
+- Configuration files (JSON, YAML, TOML, and other formats supported), including optionally watching and live-reloading.
+- Environment variables.
+- Command-line flags, primarily from `pflag.Flag` types.
+
+It is used by a wide variety of Go projects, including [hugo][hugo] and [kops][kops].
+
+[viper]: https://github.com/spf13/viper
+[hugo]: https://github.com/gohugoio/hugo
+[kops]: https://github.com/kubernetes/kops
```
Comment (mattlord) [inline content/en/docs/17.0/reference/viper/overview.md:19]: Nice. I didn't even know you could do this. 🙂

**Review by mattlord — APPROVED** [review]
❤️

File: content/en/docs/17.0/reference/viper/overview.md (hunk 5 above)
Comment (ajm188) [reply content/en/docs/17.0/reference/viper/overview.md:19]: yeah, it's part of the markdown spec! it's rad (for either super-long link targets or targets that you need to link to repeatedly)

**Pull request merged**